muscle3/muscle_manager.py

```diff
@@ -105,7 +125,13 @@ def manage_simulation(
             manager = Manager(configuration, None, log_level)
         else:
             manager = Manager(configuration, run_dir_obj, log_level)
-        print(manager.get_server_location())
+
+        server_location = manager.get_server_location()
+        if location_file is None:
+            print(server_location, flush=True)
+        else:
+            with Path(location_file).open('w') as f:
```

Comment:
You could do `Path(...).write_text(server_location)`: https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text

Reply:
Ooooh, nice, didn't know about that one. Or rather, I guess I forgot about it, because I implemented it a few years ago :smile:: https://github.com/MD-Studio/cerulean/blob/e081f0242494890e26b5ec5e150187c1f130477f/cerulean/path.py#L447